File: PE37/PE37.java
package projecteuler37;

public class PE37 {
	static boolean isPrime(int n) {
		if (n < 2) return false;
		if (n < 4) return true;
		for (int i = 2; i < (int)Math.pow(n, 0.5)+1; i++) {
			if (n % i == 0) return false;
		}
		return true;
	}
	
	static int truncateleft(int n) {
		if (n > 9) return Integer.parseInt(Integer.toString(n).substring(1));
		return 0;
	}
	
	static int truncateright(int n) {
		return n/10;
	}
	
	static boolean checkleft(int n) {
		while (n != 0) {
			if (!isPrime(n)) {
				return false;
			}
			n = truncateleft(n);
		}
		return true;
	}
	
	static boolean checkright(int n) {
		while (n != 0) {
			if (!isPrime(n)) {
				return false;
			}
			n = truncateright(n);
		}
		return true;
	}
	
	public static void main(String args[]) {
		int count = 0;
		int mysum = 0;
		int i = 9;
		while (count < 11) {
			i++;
			if (checkleft(i) && checkright(i)) {
				count++;
				mysum += i;
			}
		}
		System.out.println(mysum);
	}
}

File: PE37/PE37.py
def isPrime(n):
    if (n < 2):
        return False
    if (n < 4):
        return True
    for i in range(2, int(n**0.5)+1):
        if (n % i == 0):
            return False
    return True

def truncateleft(n):
    if (n > 9):
        return int(str(n)[1:])
    return 0

def truncateright(n):
    return n//10

def checkleft(n):
    while (n != 0):
        if (not isPrime(n)):
            return False
        n = truncateleft(n)
    return True

def checkright(n):
    while (n != 0):
        if (not isPrime(n)):
            return False
        n = truncateright(n)
    return True

count = 0
mysum = 0
i = 9
while (count < 11):
    i += 1
    if (checkleft(i) and checkright(i)):
        count += 1
        mysum += i

print(mysum)
